src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -57,7 +60,9 @@ public FunctionExecutor(
                 ConcurrencyManager concurrencyManager,
                 ILoggerFactory loggerFactory = null,
                 IEnumerable<IFunctionFilter> globalFunctionFilters = null,
-                IDrainModeManager drainModeManager = null)
+                IDrainModeManager drainModeManager = null,
+                IActivitySourceAbstraction activitySource = null,
```

Comment:
Do we need `IActivityContextProvider`? If we had just `IActivitySourceAbstraction`, we could have the implementation on the host side determine to no-op if an activity source already exists.

Reply:
Removed `IActivityContextProvider`